### DISCUSSION TITLE
TRex Cisco Issue

### DISCUSSION THREAD

**Discussion by Astrotokii** [ipspace/netlab#1536] [General]
I have a multiarea ospf set up and I want to use TRex for some realistic traffic load, but when I connect to TRex linux host and try to connect to the TRex console, I get an error connecting to RCP server. Anyone have experience using TRex docker image as linux host in netlab?

**Comment by ipspace** [comment]
When you say "I try to connect to the TRex console", what exactly do you have in mind? TRex UI trying to connect to the TRex instance? If that's the case, are you running TRex UI on the same server as the Docker containers?

As for "anyone having the experience", [nagraj-kulkarni](https://github.com/nagraj-kulkarni) got it to work, but never posted the details (more in https://github.com/ipspace/netlab/discussions/1273)

**Reply by Astrotokii** [reply]
Basically I get the node up, I netlab connect, then sudo su. I try ./trex-console command but I get an error that it cant connect to RPC server through port 4501. Im just unsure what to do to be able to connect so I can access the console.

**Reply by ipspace** [reply]
It looks like you have to have the trex server running in the container. Will try to figure out how to do that.

**Comment by ipspace** [comment]
Just to be on the safe side, could you please post a simple topology you use to run TRex?

**Reply by Astrotokii** [reply]
I will when I get home if I remember, at worst Ill post one tomorrow

**Reply by Astrotokii** [reply]
provider: libvirt

nodes:
  trx:
    device: linux
    provider: clab
    image: trexcisco/trex
  r1:
    device: iosv

links:
- group: all_links
  members: [ trx-r1, trx-r1 ]
  type: p2p

Wouldn't let me attach a .yml file, here is a basic topology. There is two links to r1 because I read that trex needs 2 links to the host its trying to send load to, one for receiving and one for sending.

**Comment by ipspace** [comment]
I couldn't get the TRex server to start automatically (and the container image does not start it). You have to connect to the container and execute **sudo bash** (to get into the right directory) followed by  **./t-rex-64 -i** (to start the traffic generator in interactive mode), then you can connect to it with trex-console from another session.

**Reply by Astrotokii** [reply]
Ohhh from another session, that makes sense. Also _sudo su_ worked to be able to access the ./t-rex-64-i as well.
